src/js/utils/RouterUtil.js

```diff
@@ -102,20 +109,33 @@ const RouterUtil = {
    */
   buildRoutes(originalRoutes) {
     let elementRoutes = RouterUtil.createComponentsFromRoutes(originalRoutes);
-    let routes = createRoutesFromReactChildren(elementRoutes[0]);
+    let routes = createRoutes(elementRoutes);
 
     return RouterUtil.setRouteConfiguration(routes, originalRoutes);
   },
 
   /**
    * Checks if a page should hide the header navigation tabs
-   * @param  {Object} router instance of react-router
+   * @param  {Array} routes instance of react-router
    * @return {Bool} should hide Page Navigation
    */
-  shouldHideNavigation(router) {
-    let routes = router.getCurrentRoutes();
-
+  shouldHideNavigation(routes) {
     return !!routes[routes.length - 1].hideHeaderNavigation;
+  },
+
+  /**
+   * Builds absolute path from routes array
+   * @param {Array} routes - an array of current routes react-rouer gives you
+   * @returns {String} path - absolute path pattern
+   */
+  reconstructPathFromRoutes(routes) {
+    let path = routes.filter(function (route) {
```

Comment:
Could you please document the current routes structure to make it easier for people to understand how this util works?